### DISCUSSION TITLE
Limit width of Column Filter Textbox

### DISCUSSION THREAD

**Discussion by rossaddison** [yiisoft/docs#223] [Show and tell]
Hi, 
I have the following in my 
````
GridView::widget()->filterCellAttributes(['style' => 'max-width: 100px;'])
````
which reduces the column width but not the filter text width so the filter is still extending over the column  into another column.
![image](https://github.com/user-attachments/assets/429444de-dc3c-4fe2-9c8f-eab264b4b127)

Any ideas how I can change this? I see there is a FilterWidget interface with 'withContent'.

**Comment by vjik** [comment]
Pass widget to `filter` parameter of column. For example:

```php
new DataColumn(
    // ...
    filter: \Yiisoft\Yii\DataView\Filter\Widget\TextInputFilter::widget()
        ->addAttributes(['style' =>'max-width: 50px']),
	// ...
),
```

**Reply by rossaddison** [reply]
Thanks Vjik. Sorted.
